### ISSUE TITLE
response shows no text

### ISSUE THREAD

**Issue by emanjavacas** [INL/BlackLab-server#4]
I am playing around with your blacklab-server and I am running against this.
A query like 
```curl "http://localhost:8080/blacklab-server-1.3.1/brown/hits?patt=%27easy%27&outputformat=json&userid=1"```

returns this (basically, the matches seem to make sense but no words nor metadata are retrieved. I tested it with the QueryTool and it does behave as expected):

```
...
"hits": [
    {
      "docPid": "7",
      "start": 282,
      "end": 283,
      "left": {
        "punct":["","","","",""],
        "pos":["","","","",""],
        "lemma":["","","","",""],
        "word":["","","","",""]
      },
      "match": {
        "punct":[""],
        "pos":[""],
        "lemma":[""],
        "word":[""]
      },
      "right": {
        "punct":["","","","",""],
        "pos":["","","","",""],
        "lemma":["","","","",""],
        "word":["","","","",""]
      }
    }
```

I also notice that in the "docInfos" field the value of "mayView" is always `false`, which sounds like relevant...
any clues?

best!

**Comment by JessedeDoes** [comment]
Hi Enrique,

Maybe this is not the issue here, but I recently had the same problem when using an older index with the latest blacklab.
Updating the index solved the problem for me.

Best
Jesse


Van: Enrique Manjavacas [mailto:notifications@github.com]
Verzonden: woensdag 8 juni 2016 20:59
Aan: INL/BlackLab-server
Onderwerp: [INL/BlackLab-server] response shows no text (#4)


I am playing around with your blacklab-server and I am running against this.
A query like
curl http://localhost:8080/blacklab-server-1.3.1/brown/hits?patt=%27easy%27&outputformat=json&userid=1

returns this (basically, the matches seem to make sense but no words nor metadata are retrieved. I tested it with the QueryTool and it does behave as expected):

...

"hits": [

    {

      "docPid": "7",

      "start": 282,

      "end": 283,

      "left": {

        "punct":["","","","",""],

        "pos":["","","","",""],

        "lemma":["","","","",""],

        "word":["","","","",""]

      },

      "match": {

        "punct":[""],

        "pos":[""],

        "lemma":[""],

        "word":[""]

      },

      "right": {

        "punct":["","","","",""],

        "pos":["","","","",""],

        "lemma":["","","","",""],

        "word":["","","","",""]

      }

    }

I also notice that in the "docInfos" field the value of "mayView" is always false, which sounds like relevant...
any clues?

best!

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/INL/BlackLab-server/issues/4>, or mute the thread<https://github.com/notifications/unsubscribe/ABDv1AWI8aoz0MRPc42gy6VAnIgPy0O2ks5qJxD_gaJpZM4IxSd3>.

________________________________

----------------------------------------------------------
Aan dit bericht kunnen geen rechten worden ontleend.
Het bericht is alleen bestemd voor de geadresseerde.
Indien het bericht niet voor u is bestemd, verzoeken wij
u dit aan ons te melden en het bericht te verwijderen.

This message shall not constitute any obligations.
This message is intended solely for the addressee.
If you have received this message in error, please
inform us and delete the message.
----------------------------------------------------------

**Comment by emanjavacas** [comment]
Hey Jesse,
thanks for the tip!

I updated blacklab to blacklab1.3.2 and reindexed the brown corpus with it, redeployed blacklab-server1.3.1 and now localhost:8080/blacklab-server-1.3.1/ shows this error:

Could not open index 'brown' (java.lang.RuntimeException: Unknown content store type fixedblock) (Internal error code 27)

best!

**Comment by jan-niestadt** [comment]
Hi Enrique,

Sorry, you're experiencing BlackLab dependency hell...The good news is that BlackLab 1.3.2 is now on Maven Central, and I've updated BlackLab Server to use this version. So if you update your clone of the BlackLab Server project to v1.3.3, which is the latest commit on master, it should from now on download the correct version of BlackLab from Maven Central and your index should work.

(The reason for that exact error is that blacklab 1.3.1 adds a new type of content store called "fixedblock" that allows us to re-use space better when deleting and re-adding documents. In retrospect, that shouldn't have been a patch update but at least a minor update - I'll try to be more careful about that in the future)

As for why you were getting empty context words before, I'm not sure. The hit context words are read from the forward index, and apparently something went wrong there. If you send your old brown corpus index, I can have a look if you want.

The 'mayView' setting determines whether users are allowed to view the entire document, and corresponds to the top-level setting ''contentViewable" in the file indexmetadata.json in your index directory. It defaults to false, so just add a top-level setting with that name and a value of true, restart the application, and you will be able to view entire documents, i.e. http://localhost:8080/blacklab-server-1.3.1/brown/docs/<id>/contents.
Hope this helps! Best,

Jan.

**Comment by jan-niestadt** [comment]
Fortunately Jesse still had his old repository. I've found and fixed the problem and sent BlackLab 1.3.3 to Maven Central. As soon as it's up, I'll update BlackLab Server to use that version as well.

**Comment by emanjavacas** [comment]
cool! thanks for you reply. I will test it this evening and let you know.

best

2016-06-09 10:42 GMT+02:00 Jan Niestadt <notifications@github.com>:

> Fortunately Jesse still had his old repository. I've found and fixed the
> problem and sent BlackLab 1.3.3 to Maven Central. As soon as it's up, I'll
> update BlackLab Server to use that version as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/INL/BlackLab-server/issues/4#issuecomment-224834551>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF6Ho6jo3pbKwweXWbj8UEVjhU8EKFb5ks5qJ9H6gaJpZM4IxSd3>
> .
>



-- 
Enrique Manjavacas.

**Comment by emanjavacas** [comment]
cool, I can confirm that the problem is solved with blacklab-server 1.3.4 and a corpus encoded with blacklab 1.3.2,

thanks!
